sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -347,29 +403,65 @@ public static Read read() {
    * sample parse function that parses click events from a table.
    *
    * <pre>{@code
-   * p.apply(BigQueryIO.read(new SerializableFunction<SchemaAndRecord, ClickEvent>() {
-   *   public Event apply(SchemaAndRecord record) {
-   *     GenericRecord r = record.getRecord();
-   *     return new Event((Long) r.get("userId"), (String) r.get("url"));
-   *   }
-   * }).from("...");
-   * }</pre>
+   * p.apply(
+   *   BigQueryIO.readAvroSource(
+   *     new SerializableFunction<SchemaAndRecord, ClickEvent>() {
+   *       public Event apply(SchemaAndRecord record) {
+   *         GenericRecord r = record.getRecord();
+   *         return new Event((Long) r.get("userId"), (String) r.get("url"));
+   *       }
+   *     }).from("..."));
+   * } </pre>
+   *
    */
-  public static <T> TypedRead<T> read(
+  public static <T> TypedRead<T> readAvroSource(
```

Comment:
Perhaps `readViaAvro` and `readViaRowProto`? - these names will be better amenable to changing to e.g. `readAllViaAvro` if we introduce a `readAll` abstraction; and "source" is a confusing and overloaded concept in Beam, whose mention in user-facing documentation is best avoided altogether.

Reply:
Done.